client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -12,35 +12,59 @@ import { connect } from 'react-redux';
  * Internal dependencies
  */
 import Card from 'components/card';
-import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
 import FormFieldset from 'components/forms/form-fieldset';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug, isJetpackMinimumVersion } from 'state/sites/selectors';
+import isJetpackModuleActive from 'state/selectors/is-jetpack-module-active';
 import isJetpackModuleUnavailableInDevelopmentMode from 'state/selectors/is-jetpack-module-unavailable-in-development-mode';
 import isJetpackSiteInDevelopmentMode from 'state/selectors/is-jetpack-site-in-development-mode';
-import { getSelectedSiteId } from 'state/ui/selectors';
-import { getSiteSlug } from 'state/sites/selectors';
+import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
 import SupportInfo from 'components/support-info';
 
 class SpeedUpSiteSettings extends Component {
 	static propTypes = {
-		fields: PropTypes.object,
 		isRequestingSettings: PropTypes.bool,
 		isSavingSettings: PropTypes.bool,
 		jetpackVersionSupportsLazyImages: PropTypes.bool,
+		submitForm: PropTypes.func.isRequired,
+		updateFields: PropTypes.func.isRequired,
 
 		// Connected props
+		assetCdnModuleActive: PropTypes.bool,
+		photonModuleActive: PropTypes.bool,
 		photonModuleUnavailable: PropTypes.bool,
 		selectedSiteId: PropTypes.number,
+		siteAcceleratorStatus: PropTypes.bool,
+		siteAcceleratorSupported: PropTypes.bool,
 		siteSlug: PropTypes.string,
 	};
 
+	handleCdnChange = () => {
+		const { siteAcceleratorStatus, submitForm, updateFields } = this.props;
+
+		// If one of them is on, we turn everything off.
+		updateFields(
+			{
+				photon: ! siteAcceleratorStatus,
+				'photon-cdn': ! siteAcceleratorStatus,
+			},
+			() => {
```

Comment:
```
() => {
	submitForm();
}
```

could be shortened to:

```
() => submitForm();
```

Reply:
Even better!

Reply:
done in d66619bd2c1568bc2d3ce50c85b1022beea0b612